code/modules/clothing/spacesuits/flightsuit.dm

```diff
@@ -439,93 +437,123 @@
 	wearer.throw_at_fast(target, (speed+density+anchored), 2, wearer)
 	wearer.visible_message("[wearer] is knocked flying by the impact!")
 
-/obj/item/device/flightpack/proc/flight_impact(atom/unmovablevictim)	//Yes, victim.
-	if(unmovablevictim == wearer)
+/obj/item/device/flightpack/proc/flight_impact(atom/unmovablevictim, crashdir)	//Yes, victim.
+	if((unmovablevictim == wearer) || crashing)
 		return FALSE
-	var/atom/movable/victim = null
-	var/dir = null
-	var/mobonly = 0
-	if(crashing)	//We're already in the process of getting knocked around by a crash.
+	crashing = TRUE
+	var/V = ""
```

Comment:
Had the problem of people crashing when they clipped a mob perpendicular to their axis of movement, this fixes that far as I can tell...